src/Console/Application.php

```diff
@@ -54,16 +54,22 @@
      */
     private $isXdebugLoaded;
 
+    /**
+     * @var bool
+     */
+    private $isUnderPhpdbg;
+
     /**
      * @var bool
      */
     private $isDebuggerDisabled;
 
-    public function __construct(Container $container, string $name = self::NAME, string $version = self::VERSION)
+    public function __construct(InfectionContainer $container, string $name = self::NAME, string $version = self::VERSION)
     {
         $this->container = $container;
-        $this->isDebuggerDisabled = ('' === trim((string) getenv(XdebugHandler::ENV_DISABLE_XDEBUG)));
-        $this->isXdebugLoaded = \extension_loaded('xdebug');
+        $this->isDebuggerDisabled = $this->container['environment.debugger.disabled'];
```

Comment:
Since our container is a `Psr\Container\ContainerInterface` (and no more), we should use `$this->container->get('environment.debugger.disabled')` instead. I guess not all PSR-11 containers implement `ArrayAccess`

Otherwise, why do we need to implement `Psr\Container\ContainerInterface`?